write-fonts/src/tables/gvar.rs

```diff
@@ -934,4 +969,119 @@ mod tests {
             Some(PackedPointNumbers::Some(vec![1, 3]))
         );
     }
+
+    fn make_39_bytes_of_variation_data() -> Vec<GlyphDeltas> {
+        vec![
+            GlyphDeltas::new(
+                Tuple::new(vec![F2Dot14::from_f32(-1.0), F2Dot14::from_f32(-1.0)]),
+                vec![
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(35, 0),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(-24, 0),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::optional(0, 0),
+                ],
+                None,
+            ),
+            GlyphDeltas::new(
+                Tuple::new(vec![F2Dot14::from_f32(1.0), F2Dot14::from_f32(1.0)]),
+                vec![
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(26, 15),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(46, 0),
+                    GlyphDelta::optional(0, 0),
+                    GlyphDelta::required(1, 0),
+                ],
+                None,
+            ),
+        ]
+    }
+
+    // sanity checking my input data for two subsequent tests
+    #[test]
+    fn who_tests_the_testers() {
+        let variations = GlyphVariations::new(GlyphId::NOTDEF, make_39_bytes_of_variation_data());
+        // to get our real size we need to mock up the shared tuples:
+        let mut tupl_map = HashMap::new();
+        tupl_map.insert(&variations.variations[0].peak_tuple, 1);
+        tupl_map.insert(&variations.variations[1].peak_tuple, 2);
+
+        let built = variations.clone().build(&tupl_map);
+        // we need an odd number for this to make sense
+        assert_eq!(built.length, 31);
+    }
+
+    #[test]
+    fn prefer_short_offsets() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        let test_data = make_39_bytes_of_variation_data();
+        let a_small_number_of_variations = (0..5)
+            .map(|i| GlyphVariations::new(GlyphId::new(i), test_data.clone()))
+            .collect();
+
+        let gvar = Gvar::new(a_small_number_of_variations).unwrap();
+        assert_eq!(gvar.compute_flags(), GvarFlags::empty());
+
+        let writer = gvar.compile_variation_data();
+        let mut sink = TableWriter::default();
+        writer.write_into(&mut sink);
+
+        let bytes = sink.into_data().bytes;
+        let expected_len = 6 * u16::RAW_BYTE_LEN // offsets: n_glyphs + 1
+                             + 32 * 5; // rounded size of each glyph, * 5 typhs
```

Comment:
is "typhs" a typo?

Reply:
um yes, that is supposed to say 'glyphs', i imagine..